maintenance_request_time_performance/models/maintenance.py

```diff
@@ -0,0 +1,151 @@
+        string='End Date and time of maintenance request completion')
+    end_request_date = fields.Date(
+        compute='_compute_end_request_date',
+        string='End Date of maintenance request completion',
+        store=True)
+    end_request_datetime_required = fields.Boolean(
+        compute='_compute_end_request_datetime_required')
+    real_request_duration = fields.Float(
+        compute='_compute_real_request_duration',
+        string='Real Duration',
+        store=True)
+    duration_difference = fields.Float(
+        compute='_compute_duration_difference',
+        string='Difference Duration',
+        store=True)
+    completion_delay = fields.Float(
+        compute='_compute_completion_delay',
+        string='Completion Delay',
+        store=True)
+
+    @api.depends('start_request_datetime')
+    def _compute_start_request_date(self):
+        for req in self:
+            req.start_request_date = \
+                (req.start_request_datetime.date()
+                 if req.start_request_datetime else None)
+
+    def _compute_start_request_datetime_required(self):
+        for req in self:
+            req.start_request_datetime_required = \
+                req.stage_id.done or req.stage_id.in_progress
+
+    @api.depends('end_request_datetime')
+    def _compute_end_request_date(self):
+        for req in self:
+            req.end_request_date = \
+                (req.end_request_datetime.date()
+                 if req.end_request_datetime else None)
+
+    def _compute_end_request_datetime_required(self):
+        for req in self:
+            req.end_request_datetime_required = req.stage_id.done
+
+    @api.depends('start_request_datetime', 'end_request_datetime')
+    def _compute_real_request_duration(self):
+        for req in self:
+            total_hours = 0
+            if req.start_request_datetime and req.end_request_datetime \
+                    and req.start_request_datetime < req.end_request_datetime:
+                diff = req.end_request_datetime - req.start_request_datetime
+                total_hours = diff.total_seconds() / 3600
+            req.real_request_duration = total_hours
+
+    @api.depends('duration', 'real_request_duration')
+    def _compute_duration_difference(self):
+        for req in self:
+            difference = 0
+            if req.duration and req.real_request_duration:
+                difference = req.real_request_duration - req.duration
+            req.duration_difference = difference
+
+    @api.depends('schedule_date',
+                 'duration',
+                 'start_request_datetime',
+                 'end_request_datetime')
+    def _compute_completion_delay(self):
+        for req in self:
+            diff = None
+            if req.end_request_datetime:
+                schedule_dt = (req.schedule_date
+                               or req.start_request_datetime
+                               or req.create_date)
```

Comment:
if `req.end_request_datetime` is filled, `req.start_request_datetime` is always filled too, isn't it? If I'm right, this line is never evaluated

Reply:
You are right. The last element of the "or" does not make sense